pre_commit/git.py

```diff
@@ -83,15 +84,16 @@ def get_files_matching(all_file_list_strategy):
     def wrapper(include_expr, exclude_expr):
         include_regex = re.compile(include_expr)
         exclude_regex = re.compile(exclude_expr)
-        return set(
-            filename
-            for filename in all_file_list_strategy()
-            if (
-                include_regex.search(filename) and
-                not exclude_regex.search(filename) and
-                os.path.lexists(filename)
+        with cwd(get_root()):  # because filenames can be relative
```

Comment:
I don't think this is necessary, pre_commit.runner.Runner.create invariantly sets the working directory (and is the first thing invoked by pre-commit)